python/cudf/tests/test_joining.py

```diff
@@ -478,3 +478,63 @@ def test_empty_joins(how, left_empty, right_empty):
     expected = left.merge(right, how=how)
     result = gleft.merge(gright, how=how)
     assert len(expected) == len(result)
+
+
+def test_merge_left_index():
+    left = pd.DataFrame({'x': [1, 2, 3, 4, 5, 6]}, index=[1, 2, 3, 4, 5, 6])
+    right = pd.DataFrame({'y': [10, 20, 30, 6, 5, 4]},
+                         index=[1, 2, 3, 4, 5, 7])
+    gleft = DataFrame.from_pandas(left)
+    gright = DataFrame.from_pandas(right)
+    pd_merge = left.merge(right, left_index=True, right_on='y')
+    gd_merge = gleft.merge(gright, left_index=True, right_on='y')
+
+    assert_eq(pd_merge, gd_merge)
+
+
+def test_merge_left_index_zero():
+    left = pd.DataFrame({'x': [1, 2, 3, 4, 5, 6]}, index=[0, 1, 2, 3, 4, 5])
+    right = pd.DataFrame({'y': [10, 20, 30, 6, 5, 4]},
+                         index=[0, 1, 2, 3, 4, 6])
+    gleft = DataFrame.from_pandas(left)
+    gright = DataFrame.from_pandas(right)
+    pd_merge = left.merge(right, left_index=True, right_on='y')
+    gd_merge = gleft.merge(gright, left_index=True, right_on='y')
+
+    assert_eq(pd_merge, gd_merge)
+
+
+def test_merge_right_index():
+    left = pd.DataFrame({'x': [1, 2, 3, 4, 5, 6]}, index=[1, 2, 3, 4, 5, 6])
+    right = pd.DataFrame({'y': [10, 20, 30, 6, 5, 4]},
+                         index=[1, 2, 3, 4, 5, 7])
+    gleft = DataFrame.from_pandas(left)
+    gright = DataFrame.from_pandas(right)
+    pd_merge = left.merge(right, right_index=True, left_on='x')
+    gd_merge = gleft.merge(gright, right_index=True, left_on='x')
+
+    assert_eq(pd_merge, gd_merge)
```

Comment:
With the exception of the index_zero test, I wonder if the rest of these could be refactored into a single parametrized test as follows:

```python
@pytest.mark.parametrize('kwargs', [
    {'left_index': True, 'right_on': 'x'},
    {'right_index': True, 'left_on': x},
    {'left_on': 'x', 'right_on': 'y'},
    {'left_index': True, 'right_index': True},
])
def test_merge_kwargs(kwargs):
    ...
    pd_merge = left.merge(right, **kwargs)
    gd_merge = gleft.merge(gright, **kwargs)

    ...
```